Merge nixos-hardware into nixpkgs under `/nixos/hardware`

Profpatsch opened the pull request:
Merges https://github.com/NixOS/nixos-hardware into nixpkgs.

This will give better discoverability, and considering the low
turnover (less than 100 commits in the last year and only 350 total)
it won’t make any dent on the size of nixpkgs.

We have a monorepo, let’s use it.

cc @Mic92 and @yegortimoshenko as the top contributors to nixos-hardware.

Followup:
- [ ] Move the few issues on nixos-hardware to the nixpkgs issue tracker
- [ ] Archive nixos-hardware
- [ ] Move the stray hardware definitions in `nixos/modules/installer/cd-dvd` (e.g. the raspberripy definitions) to `nixos/hardware`
- [ ] Think about how to integrate the tests
- [ ] Integrate into nixos-generate-config
- [ ] Mention in nixos installer configuration.

lukateras committed:
apple: enable facetimehd if allowUnfree = true

lukateras committed:
apple: wrap hardware.facetimehd option

lukateras committed:
raspberry-pi/2: format README

lukateras committed:
apple: add hid_apple.iso_layout=0 kernel param

In my experience, this makes keyboard behavior match letters printed
on it both on EU and US keyboards.

lukateras committed:
apple/macbook-air/4: add

lukateras committed:
common/pc/laptop: disable cpuFreqGovernor

lukateras committed:
apple/macbook-air/4: resolve tearing

Mic92 committed:
add lenovo x250 (#38)

Basically a meta profile that combines thinkpad + intel.

zimbatm committed:
README: add links to docs

lukateras committed:
README: add apple/macbook-air/{4,6}

lukateras committed:
CODEWONERS: init

zimbatm committed:
add Travis CI to the repo

zimbatm committed:
tests/eval-test: clean up

lukateras committed:
Merge pull request #40 from NixOS/travis-ci

lukateras committed:
tests/eval-test: Add --show-trace from #44

lukateras committed:
tests: drop -test prefix to match naming style used by NixOS tests

lukateras committed:
travis: fixup after 8d578a53d600153bc02f26cb3c80cf7bc9845ea6

lukateras committed:
travis: drop sudo

lukateras committed:
Merge pull request #45 from yegortimoshenko/20171225.193307/travis-drop-sudo

lukateras committed:
thinkpad: do not enable thinkfan by default

lukateras committed:
treewide: mark things that have to be done

makefu committed:
thinkpad/x230: init

makefu committed:
README: add x230

makefu committed:
CODEOWNERS: add makefu for x230

lukateras committed:
Merge pull request #46 from makefu/master

x230: init

lukateras committed:
CODEOWNERS: add yegortimoshenko to Lenovo ThinkPad X230

lukateras committed:
common/cpu/intel: add vaapi/vdpau drivers

I've tested this to work on MacBookAir4,1 and I'm going to deploy it
to MacBookAir6,2 today and Lenovo ThinkPad X230 in a week from now.
Also, cleaned up Lenovo ThinkPad X230 profile.

lukateras committed:
common/pc/laptop: drop explicit vm.laptop_mode

This is handled by TLP, so we don't need to set that explicitly.

lukateras committed:
LICENSE: license under CC0-1.0

lukateras committed:
apple/macbook-air/4: enable RC6p, RC6pp GPU sleep states

lukateras committed:
apple/macbook-air/6: add acpi_osi= kernel param

lukateras committed:
apple/macbook-air/6: resolve tearing

lukateras committed:
lenovo/thinkpad/x230: resolve tearing

lukateras committed:
Merge pull request #47 from yegortimoshenko/20171227.165607/cc-1.0

LICENSE: license under CC0-1.0

not-fl3 committed:
default.nix, fix for nix-env -qa

lukateras committed:
Merge pull request #53 from not-fl3/master

default.nix at repo root, for nix-env -qa

lukateras committed:
pcengines/apu: init

yorickvP committed:
add XPS 13 (9360)

lukateras committed:
dell/xps/13-9360: review

lukateras committed:
Merge pull request #58 from yorickvP/master

add XPS 13 (9360)

lukateras committed:
qca6174-firmware: pin down commit

yorickvP committed:
common/cpu/intel/kaby-lake: fix kernelParams -> boot.kernelParams

lukateras committed:
Merge pull request #59 from yorickvP/patch-1

common/cpu/intel/kaby-lake: fix kernelParams -> boot.kernelParams

azazel75 committed:
Add basic support for 6th-gen X1

azazel75 committed:
Give TLP more control over the battery and enable battery life saver parameters

azazel75 committed:
Move battery settings to the configuration for all the 6th gen models

azazel75 committed:
Add the temporary fix for the cpu-throttling issue

See https://wiki.archlinux.org/index.php/Lenovo_ThinkPad_X1_Carbon_(Gen_6)#Power_management.2FThrottling_issues

azazel75 committed:
Add stub for the S3 suspend state issue

See https://delta-xi.net/#056

azazel75 committed:
Add a bit of documentation

azazel75 committed:
Cosmetic

azazel75 committed:
Set a default to fix GTK/GNOME ui elements appareance

michaelpj committed:
Add acpi_call module

michaelpj committed:
Add basic t480s profile

lukateras committed:
Merge pull request #61 from michaelpj/imp/t480s

Add a basic t480s profile

lukateras committed:
Add Purism Librem 13v3 profile

lukateras committed:
Add PC Engines APU to the table

lukateras committed:
Add ThinkPad X270 profile

commandodev committed:
Add a note about rcu_sched problems

How I fixed my XPS15 after it being dead for two months

Commit:
Add a link to the other forum thread I found

lukateras committed:
Merge pull request #63 from boothead/patch-1

Add a note about rcu_sched problems

ashgillman committed:
Add Dell E7240 Profile

(Current XPS 15 works well)

lukateras committed:
Fix tests

michaelpj committed:
Suggestions for CONTRIBUTING

lukateras committed:
Merge pull request #66 from NixOS/fix-tests

Fix tests

azazel75 committed:
Add optional console font configuration to make it readable on QHD displays

azazel75 committed:
Fix typo

azazel75 committed:
Add missing pkgs parameter

azazel75 committed:
Enable powersave features when on battery (taken from #60)

rehno-lindeque committed:
apple/macbook-pro/11-5: init

rehno-lindeque committed:
apple/macbook-pro/11-5: Additional wiki resources

As suggested by issue #41.

rehno-lindeque committed:
apple/macbook-pro/11-5: Change README to markdown

.wiki is just plain distracting to edit, markdown allows me to quickly write
up accompanying docs.

rehno-lindeque committed:
apple/macbookpro/11-5: Prevent intermittent xhci wakeup after suspend

Mic92 committed:
Merge pull request #60 from azazel75/master

Add basic support for 6th-gen X1

azazel75 committed:
Link X1 6th gen to the README (#69)

rehno-lindeque committed:
apple/macbook-pro/11-5: Remove boot loader

Since the boot loader is already automatically generated by
nixos-generate-config, there is no need to include it here.

Mic92 committed:
Merge pull request #68 from rehno-lindeque/apple/macbookpro/11-5/init

apple/macbookpro/11-5: init

lukateras committed:
intel: videoDrivers: intel -> modesetting (#57)

rvolosatovs committed:
lenovo/thinkpad/x260: Init

michaelpj committed:
T480s: use CPU throttling fix too

rehno-lindeque committed:
apple/macbookpro/11-5: Add to tests

Mic92 committed:
Merge pull request #72 from rehno-lindeque/apple/macbookpro/11-5/test

apple/macbookpro/11-5: Add to tests

rvolosatovs committed:
default: Add Lenovo x260

Mic92 committed:
Merge pull request #71 from michaelpj/imp/t480s-throttling

T480s: use CPU throttling fix too

lukateras committed:
Add Toshiba swanky along with documentation

lukateras committed:
Add to README.md, redo the table in Markdown

lukateras committed:
Toshiba Chromebook (`swanky`) -> Toshiba Chromebook 2 `swanky`

lukateras committed:
toshiba/swanky: reword guide

lukateras committed:
CONTRIBUTING.org -> CONTRIBUTING.md

lukateras committed:
Add toshiba-swanky to CI

lukateras committed:
toshiba/swanky: fix eval

lukateras committed:
Merge pull request #73 from NixOS/20180824.021016/swanky

Add Toshiba Chromebook 2 (swanky), Org -> Markdown

lukateras committed:
LICENSE -> COPYING

lukateras committed:
cpu/intel: early KMS

lukateras committed:
Merge pull request #75 from NixOS/20180825.135451/early-kms

cpu/intel: early KMS

rvolosatovs committed:
README: Add Lenovo ThinkPad x260

rvolosatovs committed:
lenovo/thinkpad/x260: Remove opinionated TLP config

erictapen committed:
thinkpad: introduce t430

Copied everything from t440p, added a fix for the brightness keys

Mic92 committed:
Merge pull request #77 from Mic92/imp/contributing-suggestions

[rebased from #67] Suggestions for CONTRIBUTING

erictapen committed:
README: add Lenovo Thinkpad T430

azazel75 committed:
Fix cpu-throttling service file reference

azazel75 committed:
Fix acpi override load

As proposed by @garbas in #60

azazel75 committed:
Add X1 profile to the README.md

azazel75 committed:
Unfortunately all the table has to be modified to have the changes rendered correctly

azazel75 committed:
Fix table rendering. Add the target of the link to the X1 profile

rvolosatovs committed:
lenovo/thinkpad/x260: Use `med_power_with_dipm` in TLP

lukateras committed:
ThinkPad X1 gen6 -> ThinkPad X1 (6th Gen)

lukateras committed:
Merge pull request #78 from azazel75/up

Fix unusable Lenovo Thinkpad X1 profile

Mic92 committed:
Merge pull request #70 from rvolosatovs/feature/x260

lenovo/thinkpad/x260: Init

Mic92 committed:
Merge pull request #76 from erictapen/t430-init

thinkpad: introduce t430

lukateras committed:
beagleboard/pocketbeagle: init

lukateras committed:
CODEOWNERS: add yegortimoshenko to three devices

- BeagleBoard PocketBeagle
- PC Engines APU
- Purism Librem 13v3

asymmetric committed:
lenovo/thinkpad/t450s: init

Mic92 committed:
Merge pull request #80 from asymmetric/asymmetric/t450s

lenovo/thinkpad/t450s: init

Moredread committed:
dell/xps/13-9370: init

Moredread committed:
CODEOWNERS: add moredread for Dell XPS 13 9370

Moredread committed:
move acpi_call.nix and cpu-throttling-bug.nix to common folder

Mic92 committed:
Merge pull request #82 from NixOS/travis-test-build

copy of #81

lukateras committed:
common/cpu: default microcode updates to enableRedistributableFirmware

lukateras committed:
Merge pull request #85 from NixOS/201810/microcode

common/cpu: default microcode updates to enableRedistributableFirmware

schmittlauch committed:
add basic support for Lenovo Thinkpad T440s (#83)

This adds basic support for the Thinkpad T440s, based on the work done for the T440p. The expression enables:

- microcode updates
- trackpoint support
- kernel module for battery level management

bhipple committed:
lenovo x1 carbon gen6: update S3 suspend

With UEFI >= 1.30, there's an explicit option to enable S3 power management.
Once this is selected, S3 is enabled and "deep" is selected as the default
/sys/power/mem_sleep value without requiring any kernel boot parameters.

lukateras committed:
Merge pull request #88 from bhipple/update/x1-carbon

lenovo x1 carbon gen6: update S3 suspend

Mic92 committed:
fix nix-env evaluation

nix-env picked up `default.nix`, which contained our hardware profiles.
This is not only cpu-intensive to evaluate but also fails because
it does some import-from-derivation.
By moving default.nix to release.nix it should be no longer loaded

Mic92 committed:
Merge pull request #92 from NixOS/fix-nix-env

fix nix-env evaluation

Mic92 committed:
add dummy default.nix

lukateras committed:
Merge pull request #95 from NixOS/nix-env-part2

add dummy default.nix

kalbasit committed:
dell/xps/13-9380: init (#97)

This is pretty much a copy of XPS 9370, without kaby-laky and the throttle bug. I was getting lockups with the kaby-laky changes and the throttle bug did not seem to have an impact either way.

eyJhb committed:
lenovo-x1: removed battery TLP threshholds

lukateras committed:
Merge pull request #100 from eyJhb/lenovo-x1-thresh

lenovo-x1: removed battery TLP threshholds

tomfitzhenry committed:
pcengines/apu: expose GRUB over serial

Prior to this commit, an apu's boot would go through stages of varying visibility:
1. BIOS: exposed to serial
2. GRUB: not exposed to serial
3. Linux: exposed to serial

This commit changes (2) to be visible over serial, so the entire boot is visible over serial.

Tested on nixos 18.09 on a https://www.pcengines.ch/apu1d.htm

Mic92 committed:
Merge pull request #103 from tomfitzhenry/patch-1

pcengines/apu: expose GRUB over serial

angristan committed:
README: Add Dell XPS 13 9360 profile (#104)

And reorder other XPS 13.

jn3to committed:
t430: add acpi_call module

JorisE committed:
Add thinkpad 480s to list of profiles.

JorisE committed:
Add basic support for the thinkpad t470s

Mic92 committed:
Merge pull request #107 from JorisE/master

Added basic support for the t470s

Mic92 committed:
Merge pull request #106 from jnetod/master

t430: add acpi_call module

Mic92 committed:
dell/xps/13-9380: also import intel profile

Mic92 committed:
dell/xps/13-9380: mention fwupd

Mic92 committed:
Merge pull request #110 from NixOS/xps-13

dell/xps/13-9380: also import intel profile

Mic92 committed:
toshiba/swanky: switch to default kernels

Both unstable and 19.03 come with 4.19+ kernels
making this option obsolete.

Commit:
Specify a correct unit description

Mic92 committed:
Specify a correct unit description (#112)

Specify a correct unit description

tomberek committed:
Initial 15-7590

michaelpj committed:
Replace ad hoc throttling fix with throttled

`throttled` is a service designed to fix the same CPU throttling bug,
but it's an actively maintainted upstream project that we can use.

tomberek committed:
Wifi fixed by removing iwlwifi-cc-a0-48.ucode

Mic92 committed:
enable fstrim on SSDs

This will run TRIM once a week using a systemd timer. Running TRIM regularly
will improves the performance and increases the SSDs lifespan. Since it is based
on the utillinux no additional package is required.

lukateras committed:
Merge pull request #116 from NixOS/fstrim

enable fstrim on SSDs

Mic92 committed:
Merge pull request #111 from NixOS/cleanup-swanky

toshiba/swanky: switch to default kernels

tomberek committed:
Update dell/xps/15-7590/README.wiki

Co-Authored-By: Jörg Thalheim <Mic92@users.noreply.github.com>

tomberek committed:
Suggested fixes

Ma27 committed:
thinkpad: emulate wheel for a trackpoint by default

Most trackpoint users I know use the middle-button to have some
scrolling functionality. So I think that for ThinkPads at least we
should have this enabled by default if the trackpoint shall be used.

Ma27 committed:
thinkpad/x1/6th-gen: fix trackpoint name

Newer ThinkPads have a new name for the Trackpoint - "TPPS/2 Elan TrackPoint".

Having the "[...] IBM Trackpoint" as default caused some weird
side-effects on my machine (e.g. stopped the trackpoint working after a
suspend) with the wrong name. Although this is mentioned in the option's
description[1], I think that this should be declared explicitly here.

As soon, as we supported newer models as well[2], we should probably
move this into its own common profile.

[1] https://nixos.org/nixos/options.html#hardware.trackpoint.device
[2] https://certification.ubuntu.com/catalog/component/input/5313/input%3ATPPS/2ElanTrackPoint/

exi committed:
Add better pulseaudio support for thinkpad t490

SRGOM committed:
Update README.md

At first (till third glance) I thought only the listed profiles are supported

exi committed:
update readme for T490

globin committed:
Merge pull request #118 from exi/master

Add better pulseaudio support for thinkpad t490

globin committed:
Merge pull request #117 from Ma27/thinkpad-trackpoint-fixes

thinkpad: minor trackpoint fixes

rycee committed:
thinkpad/x250: add acpi_call

This kernel module is needed to, for example, support tlp's

  - `START_CHARGE_THRESH_BAT?` and
  - `STOP_CHARGE_THRESH_BAT?`

options.

Mic92 committed:
Merge pull request #120 from rycee/x250/acpi_call

thinkpad/x250: add acpi_call

Mic92 committed:
Merge pull request #119 from SRGOM/patch-1

Update README.md

rycee committed:
thinkpad/x1-extreme/gen2: add module

exi committed:
match specifically on the sound chip

globin committed:
Merge pull request #124 from exi/master

match specifically on the sound chip

Mic92 committed:
Merge pull request #123 from rycee/thinkpad-x1-extreme-gen2

thinkpad/x1-extreme/gen2: add module

Mic92 committed:
Merge branch 'master' into add-e7240

Mic92 committed:
Add Dell E7240 Profile (#122)

Add Dell E7240 Profile

adamse committed:
Fix readme markdown syntax

Mic92 committed:
Merge pull request #126 from adamse/patch-2

Fix readme markdown syntax

rummik committed:
Initial p53 hardware configuration (#125)

Mic92 committed:
Merge pull request #115 from michaelpj/imp/throttled

Replace ad hoc throttling fix with throttled

asymmetric committed:
intel: add intel-media-driver

Provides hardware accelerated video playback from Broadwell onwards.

disassembler committed:
dell/xps: enable thermald service

domenkozar committed:
Merge pull request #127 from NixOS/dell-thermald

dell/xps: enable thermald service

Alecktos committed:
Add Dell XPS-7390 Profile (#128)

Br1ght0ne committed:
Add Lenovo ThinkPad T420 profile

ilian committed:
apple: Add support for MacBook Air 3,X (#91)

Alecktos committed:
Add deepsleep to avoid wakeups during sleep mode (#136)

mishudark committed:
Add Lenovo ThinkPad X280 profile

Mic92 committed:
Merge pull request #137 from mishudark/x280

Add Lenovo ThinkPad X280 profile

rummik committed:
Add Lenovo ThinkPad L13 profile

rummik committed:
Update README.md

tomberek committed:
Make note about EFI and remove 5.1 kernel hardcode

tomberek committed:
Apply suggestions from 9999years

Mic92 committed:
Merge pull request #114 from tomberek/master

Initial 15-7590

Mic92 committed:
README.md: link to dell xps 15-7590

Mic92 committed:
xps-15-7590: convert wiki to markdown.

This allows it to symlink it in the README

Mic92 committed:
dell-xps-15-7590: fix README link

Mic92 committed:
Revert "xps-15-7590: convert wiki to markdown."

This reverts commit 77983d16f6ddcbdaf65d7656e82d39669095be6f.

apparently github can link to wiki text fine.

danielstaleiny committed:
Add xps 9560 with its own quirks, I added versions:
1 default one, with nvidia working and intel working, needs optirun to
run application with nvidia
2 intel only, I am using this as my daily driver for long time, only
intel is turned on and nvidia is turned off and doesn't drain any
power.
3 nvidia only, intel is dissabled and nvidia is picked as default gpu,
no need to fiddle with optirun etc, good for gpu heavy application.

All these settings needs to reboot to work properly

danielstaleiny committed:
add xps 9560 root readme to be able to easily find

9999years committed:
Add Dell XPS 15-7590 suggested options

These suggested options are taken from my configuration but aren't quite
universal enough to be enabled by default.

Mic92 committed:
xps/15-7590: don't enable unfree firmware by default

Also don't set nixpkgs.pkgs. Instead just add an overlay.

Mic92 committed:
Merge pull request #142 from 9999years/7590

Add Dell XPS 15-7590 suggested options

Mic92 committed:
Merge pull request #138 from rummik/add-lenovo-thinkpad-l13

Add Lenovo ThinkPad L13 profile

Qubasa committed:
Added profile for issues 69289

Qubasa committed:
Update lenovo/thinkpad/e495/default.nix

Co-Authored-By: Jörg Thalheim <Mic92@users.noreply.github.com>

Qubasa committed:
Added entry Thnikpad E495 to README

Mic92 committed:
Merge pull request #144 from Luis-Hebendanz/master

Added profile for issues 69289

Mic92 committed:
thinkpad/e495: fix evaluation

fixes #145

Mic92 committed:
Merge pull request #146 from Mic92/e495

thinkpad/e495: fix evaluation

Mic92 committed:
Merge pull request #143 from Mic92/xps-15-7590

xps/15-7590: don't enable unfree firmware by default

9999years committed:
Support NixOS 19.09 for <dell/xps/15-7590> (#141)

Virtual console options were renamed in 20.03; use `console.earlySetup`
or `boot.earlyVconsoleSetup` depending on OS version.

https://github.com/NixOS/nixos-hardware/pull/114#discussion_r374953204

Mic92 committed:
Merge pull request #139 from danielstaleiny/master

Adding Dell XPS 15 9560

Mic92 committed:
Add parallel test runner

The new test runner will evaluate all test profiles from the README.md
in parallel in separate nix-build processes.
Since we do not load all processes into one process,
this also helps saving memory.

Mic92 committed:
Update README.wiki

danderson committed:
lenovo/thinkpad/t495: init from e495.

The E495 and T495 are based on the same core Ryzen platform, so the
E495 tweaks apply similarly to the T495.

Tested on my own T495, works well.

Mic92 committed:
Merge pull request #147 from Mic92/fix-travis

travis: parallelize evaluation and limit RAM usage

Mic92 committed:
Merge pull request #149 from danderson/master

lenovo/thinkpad/t495: init from e495.

danderson committed:
thinkpad/t495: fix backlight save/load and and battery conditioning.

Saving/loading the backlight state requires the acpi_backlight=native
kernel parameter.

acpi_call is required by TLP to get access to battery conditioning
information from the firmware.

Signed-off-by: David Anderson <dave@natulte.net>

Mic92 committed:
Merge pull request #150 from danderson/master

thinkpad/t495: fix backlight save/load and and battery conditioning.

ngiger committed:
Document librem laptop 15v3

ngiger committed:
Automatically lock the desktop when removing the librem key

ngiger committed:
Create separate directories for librem 13v3 and 15v3

Mic92 committed:
Merge pull request #152 from ngiger/librem15v3

Added Librem15v3

DamienCassou committed:
lenovo/thinkpad/t460s: Remove problematic options

See there links for details:

https://github.com/NixOS/nixos-hardware/pull/134#pullrequestreview-336181593
https://github.com/NixOS/nixos-hardware/pull/134#discussion_r361146814

DamienCassou committed:
lenovo/thinkpad/t460s: Add missing acpi_call module

This is required by tlp for battery management.

Mic92 committed:
Merge pull request #153 from DamienCassou/fix-t460s

Improve support for t460s

j0hax committed:
Change TLP charge thresholds

Mic92 committed:
Merge pull request #154 from j0hax/x230-tlp-thresholds

Mic92 committed:
Revert "Change TLP charge thresholds"

This reverts commit 76f681355cb6c2aab79d190907fbb521bd19ec87.

This setting is opinionated and should be left to the user.
See: https://github.com/NixOS/nixos-hardware/pull/154#issuecomment-617181517

Mic92 committed:
Merge pull request #155 from Mic92/remove-thresholds

wmertens committed:
7390: clarify intent

wmertens committed:
wiki syntax

delan committed:
README: fix nix-channel --update command

Following the current instructions yields these errors for some reason:

```
$ sudo nixos-rebuild switch
warning: Nix search path entry '/nix/var/nix/profiles/per-user/root/channels/nixos' does not exist, ignoring
error: file 'nixpkgs/nixos' was not found in the Nix search path (add it using $NIX_PATH or -I), at (string):1:13
building Nix...
warning: Nix search path entry '/nix/var/nix/profiles/per-user/root/channels/nixos' does not exist, ignoring
error: file 'nixpkgs/nixos' was not found in the Nix search path (add it using $NIX_PATH or -I)
warning: Nix search path entry '/nix/var/nix/profiles/per-user/root/channels/nixos' does not exist, ignoring
error: file 'nixpkgs' was not found in the Nix search path (add it using $NIX_PATH or -I)
warning: Nix search path entry '/nix/var/nix/profiles/per-user/root/channels/nixos' does not exist, ignoring
error: file 'nixpkgs/nixos/modules/installer/tools/nix-fallback-paths.nix' was not found in the Nix search path (add it using $NIX_PATH or -I)
/tmp/nixos-rebuild.vs4JVw/nix
warning: Nix search path entry '/nix/var/nix/profiles/per-user/root/channels/nixos' does not exist, ignoring
error: file 'nixpkgs' was not found in the Nix search path (add it using $NIX_PATH or -I)
building the system configuration...
warning: Nix search path entry '/nix/var/nix/profiles/per-user/root/channels/nixos' does not exist, ignoring
error: file 'nixpkgs/nixos' was not found in the Nix search path (add it using $NIX_PATH or -I)
```

Mic92 committed:
Merge pull request #156 from delan/patch-1

Commit:
New dell/latitude/3480

The touchpad is painful to use unless the "psmouse" kernel module
is disabled.

Mic92 committed:
Merge pull request #158 from ehmry/master

matthiasbeyer committed:
Add InfinityBook v4 (#159)

Mic92 committed:
Merge pull request #131 from asymmetric/imd

alyssais committed:
google/pixelbook: init

Mic92 committed:
Merge pull request #161 from alyssais/pixelbook

00-matt committed:
dell/xps/13-9360: Stop using i18n.consoleFont

trace: warning: The option `i18n.consoleFont' defined in
`/nix/var/nix/profiles/per-user/root/channels/nixos-hardware/dell/xps/13-9360'
has been renamed to `console.font'.

Mic92 committed:
Merge pull request #163 from 00-matt/dell-xps-13-9360/consolefont

phunehehe committed:
lenovo/thinkpad/x1/7th-gen: init

Mic92 committed:
make tests less verbose

Mic92 committed:
Merge pull request #165 from Mic92/quiet-tests

Mic92 committed:
README.md: add missing link

Mic92 committed:
Merge pull request #164 from phunehehe/x1

Mic92 committed:
Merge pull request #166 from Mic92/cleanup-readme

Mic92 committed:
fix toshiba/swanky evaluation (#167)

Mic92 committed:
tests: hide evaluation command behind verbose flag

erikarvstedt committed:
pc/laptop: remove redundant cpuFreqGovernor setting

This setting has been part of NixOS since version 16.09

Mic92 committed:
Merge pull request #168 from Mic92/tests

Mic92 committed:
Merge pull request #169 from erikarvstedt/improve-laptop-preset

Profpatsch committed:
nixos-hardware: prepare for merge by moving to nixos/hardware

It is going to be merged into nixpkgs, under `nixos/hardware`.

This will give better discoverability, and considering the low
turnover (less than 100 commits in the last year and only 350 total)
it won’t make any dent on the size of nixpkgs.

We have a monorepo, let’s use it.

Profpatsch committed:
Merge branch 'nixos-hardware-merge-preparations' into nixos-hardware-merge

Merges https://github.com/NixOS/nixos-hardware into nixpkgs.

This will give better discoverability, and considering the low
turnover (less than 100 commits in the last year and only 350 total)
it won’t make any dent on the size of nixpkgs.

We have a monorepo, let’s use it

Profpatsch committed:
nixos/hardware: move CODEOWNERS to toplevel

Profpatsch committed:
nixos/hardware: remove .gitignore

It’s a full subset of the toplevel .gitignore.

yegortimoshenko commented on nixos/hardware/.travis.yml:
```diff
@@ -0,0 +1,3 @@
+language: nix
```
Would make sense to have this file removed, I think.

Profpatsch replied:
Yeah, we can think about how to integrate it into the existing tests later.

yegortimoshenko commented on nixos/hardware/COPYING:
```diff
@@ -0,0 +1,111 @@
+Statement of Purpose
```
This is a public domain license. So, for simplicity's sake, I'd remove it and keep a top-level MIT license.

Profpatsch replied:
I wasn’t sure of that, as IANAL I opted for keeping it safe. I would make that decision in a separate PR.

lukateras reviewed (approved):
I think this is a great idea.

Profpatsch committed:
nixos/hardware: remove .travis.yml

We are going to have to integrate the tests differently, but it’s fine
for now.

Mic92 commented:
I am also all for it. It also makes it easier to apply changes that depend on specific nixpkgs revisions.

Mic92 commented on .github/CODEOWNERS:
```diff
@@ -55,6 +55,15 @@
 
+# NixOS hardware descriptions
+/nixos/hardware/beagleboard/pocketbeagle @yegortimoshenko
+/nixos/hardware/dell/xps/13-9370 @moredread
```
cc @Moredread

Mic92 commented on .github/CODEOWNERS:
```diff
@@ -55,6 +55,15 @@
+# NixOS hardware descriptions
+/nixos/hardware/beagleboard/pocketbeagle @yegortimoshenko
+/nixos/hardware/dell/xps/13-9370 @moredread
+/nixos/hardware/dell/xps/13-9380 @kalbasit
```
@kalbasit

grahamc commented:
I don't think this is a good idea. These profiles aren't tested regularly, and merging them in to nixpkgs means there should be a quality standard IMO. They're also fairly opinionated in ways that our users may not want. I don't this is a healthy move for Nixpkgs or for the nixos-hardware project.`

grahamc commented:
I'm also not sure about the license. Nixpkgs' expressions are all MIT. I think this should be reverted for more consideration.

edolstra commented:
Indeed, changes to the scope of Nixpkgs require an RFC.

cleverca22 commented:
seems this can also sometimes confuse `git bisect`:
```
[clever@amd-nixos:~/apps/nixpkgs]$ git bisect bad
Bisecting: 284 revisions left to test after this (roughly 8 steps)
[c3b2ac63ff06e5dd4fa66f75bfab7892b3edfdc5] Add deepsleep to avoid wakeups during sleep mode (#136)

[clever@amd-nixos:~/apps/nixpkgs]$ nix-build -A platformio && ls -l result/
attribute 'platformio' in selection path 'platformio' not found

[clever@amd-nixos:~/apps/nixpkgs]$ nix repl .
Welcome to Nix version 3.0pre20200829_f156513. Type :? for help.

Loading '.'...
Added 0 variables.

nix-repl> :q

[clever@amd-nixos:~/apps/nixpkgs]$ cat default.nix
# This file is necessary so nix-env -qa does not break,
# when nixos-hardware is used as a channel
{}

[clever@amd-nixos:~/apps/nixpkgs]$
```

bisect randomly switched over to commits from nixos-hardware, and then `nix-build` fails due to an empty set
